src/main/java/fact/extraction/SinglePulseExtraction.java

```diff
@@ -148,6 +170,42 @@ private void addStartSliceOffset(int[][] arr) {
         }
     }
 
+    public double[] loadIntegralGainFile(SourceURL inputUrl, Logger log) {
+        factSinglePePulseIntegral = TemplatePulse.factSinglePePulseIntegral();
```

Comment:
this is shit .. I am misusing loadIntegralGainFile as an `__init__` method. so I initialize this member here...
but that's of course total bullshit.